File: JavaPerformanceTest/src/Database_Performance_Test.java
import java.sql.*;
import java.util.LinkedList;
import java.util.List;

class Database_Performance_Test {
    static class UnicodeEntry {
        private int ID;
        private char UnicodeCharacter;
        private int PREVIOUS;
        private int NEXT;

        UnicodeEntry(int ID, char UnicodeCharacter, int PREVIOUS, int NEXT) {
            this.ID = ID;
            this.UnicodeCharacter = UnicodeCharacter;
            this.PREVIOUS = PREVIOUS;
            this.NEXT = NEXT;
        }

        public int getID() {
            return ID;
        }

        public void setID(int ID) {
            this.ID = ID;
        }

        public char getUnicodeCharacter() {
            return UnicodeCharacter;
        }

        public void setUnicodeCharacter(char unicodeCharacter) {
            UnicodeCharacter = unicodeCharacter;
        }

        public int getPREVIOUS() {
            return PREVIOUS;
        }

        public void setPREVIOUS(int PREVIOUS) {
            this.PREVIOUS = PREVIOUS;
        }

        public int getNEXT() {
            return NEXT;
        }

        public void setNEXT(int NEXT) {
            this.NEXT = NEXT;
        }
    }

    void executeTest(){
        List<UnicodeEntry> unicodeEntryList = new LinkedList<>();
        try (Connection connection = DriverManager.getConnection("jdbc:sqlite:unicode.sqlite")) {
            Statement statement = connection.createStatement();
            ResultSet resultSet = statement.executeQuery("SELECT * FROM unicode;");
            while (resultSet.next()){
                String resultChar = resultSet.getString("CHAR");
                char CHAR = resultChar.length() > 0 ? resultChar.charAt(0) : 0;
                unicodeEntryList.add(
                        new UnicodeEntry(
                                resultSet.getInt("ID"),
                                CHAR,
                                resultSet.getInt("PREVIOUS"),
                                resultSet.getInt("NEXT")
                        )
                );
            }
        } catch (SQLException throwables) {
            throwables.printStackTrace();
        }
    }
}

File: PerformanceTest/Scala/Database_Performance_Test.scala
import java.sql.{Connection, DriverManager, ResultSet, SQLException, Statement}
import scala.+:
import scala.collection.LinearSeq
import scala.collection.mutable.ListBuffer

class Database_Performance_Test {
  class UnicodeEntry(var ID: Int, var UnicodeCharacter: Char, var PREVIOUS: Int, var NEXT: Int) {
    def getID: Int = ID

    def setID(ID: Int): Unit = {
      this.ID = ID
    }

    def getPREVIOUS: Int = PREVIOUS

    def setPREVIOUS(PREVIOUS: Int): Unit = {
      this.PREVIOUS = PREVIOUS
    }

    def getNEXT: Int = NEXT

    def setNEXT(NEXT: Int): Unit = {
      this.NEXT = NEXT
    }

    def getUnicodeCharacter: Char = UnicodeCharacter

    def setUnicodeCharacter(UnicodeCharacter: Char): Unit = {
      this.UnicodeCharacter = UnicodeCharacter
    }
  }

  def execute(): Unit = {
    var unicodeEntryList = new ListBuffer[UnicodeEntry]
    try {
      val connection: Connection = DriverManager.getConnection("jdbc:sqlite:unicode.sqlite")
      val statement: Statement = connection.createStatement()
      val resultSet: ResultSet = statement.executeQuery("SELECT * FROM unicode")
      while (resultSet.next()) {
        val resultChar: String = resultSet.getString("CHAR")
        val CHAR: Char = if (resultChar.nonEmpty) resultChar.charAt(0) else ' '
        unicodeEntryList += new UnicodeEntry(
          ID = resultSet.getInt("ID"),
          UnicodeCharacter = CHAR,
          PREVIOUS = resultSet.getInt("PREVIOUS"),
          NEXT = resultSet.getInt("NEXT")
        )
      }
    }
    catch{
      case e: SQLException => e.printStackTrace()
    }
  }
}
